{{ cookiecutter.project_slug }}/{{ cookiecutter.project_slug }}/apps/common/formatters.py

```diff
@@ -0,0 +1,25 @@
+from pythonjsonlogger import jsonlogger
+
+
+class DjangoRequestJsonFormatter(jsonlogger.JsonFormatter):
+    """
+    This class is used to serialize headers and body of the JSON Django request
+    into the `request` property in the record of `jsonlogger.JsonFormatter`.
+    """
+
+    def add_fields(self, log_record, record, message_dict):
+        if getattr(record, 'request', None):
+            def is_django_request(request):
+                return getattr(record.request, 'method', None)
+
+            def is_json_request(request):
+                return record.request.META.get('CONTENT_TYPE', None) == 'application/json'
```

Comment:
@maticomp what about meeting half way and including the `application/x-www-form-urlencoded` along with `application/json`?

Reply:
@Struchu fair enough, that'd be useful too. It's okay as long as multiparts are excluded.

Reply:
Cool. @kaniak274 you heard the master ;)

Reply:
Yeah I did :)

Reply:
One more: does the current logging configuration log the source (and forwarded IP)? It would be quite useful too.

Reply:
It should be included in the `headers` if the proxy is properly setting it up.

Reply:
Plus logger from the lib uses two types of modules to log events one associated with django request and second for `basehttp` module which looks like this:
```
{"levelname": "ERROR", "module": "basehttp", "process": 119, "thread": 140205281494784, "message": "\"GET /test/ HTTP/1.1\" 500 85017", "request": "<socket.socket fd=6, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=0, laddr=('172.26.0.3', 8000), raddr=('172.26.0.1', 49750)>", "server_time": "18/Jun/2020 08:02:53", "status_code": 500}
```

Reply:
Ah, okay. Well, that should do for now then, we'll revisit later. 

For me, as soon as body is decoupled from headers, this can be merged.